src/index.ts

```diff
@@ -172,6 +172,12 @@ export class AwaitQueue
 		// Execute it.
 		await this.executeTask(pendingTask);
 
+		// If the task is stopped, ignore it.
+		if (pendingTask.stopped)
+		{
+			return;
```

Comment:
I think this will make the queue get stuck in case another task B was added while A was executing if remove(0) is called so task A gets stopped.